(feat): Agent Control (#19943)
newrelic/docs-website: src/content/docs/new-relic-control/fleet-control/overview.mdx

---
title: "Overview of Fleet Control"
metaDescription: "Learn about Fleet Control's centralized remote management of Kubernetes clusters with New Relic agents, overseen by Agent Control"
freshnessValidatedDate: never
---

<Callout title="preview">
  We're still working on this feature, but we'd love for you to try it out!

  This feature is currently provided as part of a preview program pursuant to our [pre-release policies](/docs/licenses/license-information/referenced-policies/new-relic-pre-release-policy).
</Callout>

> Fleet Control provides centralized and remote command over fleets of managed clusters for full instrumentation lifecycle management at scale. Use Fleet control to:

Reviewer: ```suggestion
Fleet Control provides centralized and remote command over fleets of managed clusters for full instrumentation lifecycle management at scale. Use Fleet Control to:
```

- **Centralize agent operations and reduce manual toil:** Install, monitor, configure, and update all agents and integrations in one place.
- **Customize fleets for bulk management:** Group agents by instrumentation needs and easily upgrade agent versions for entire fleets with a few clicks. No scripts needed.
- **Eliminate telemetry gaps:** Manage agents on Kubernetes clusters with one lightweight supervisor—Agent Control—that scales across your environment.

> 

Reviewer 2: In this space please add this screenshot of a fleet: https://drive.google.com/file/d/1q8ZiJblJ_dehSvOKdib453PPanwk-gY_/view?usp=sharing

Author: Can you please share individual screenshots? I am not sure we should use the orange background in our docs. Also, I couldn't find my way to the Agents in this fleet page.

> To navigate to Fleet Control, go to **Capabilities → New Relic Control → Fleets**. Below is an example of a single fleet:

Reviewer 3: ```suggestion
To navigate to Fleet Control, go to **Capabilities → New Relic Control → Fleets**.  You can use the pin to add New Relic Control to the left side navigation bar for easy access.
```
Insert the following image: https://drive.google.com/file/d/1mq8oEi9fyljMZQsUrnADF8nDchO3I9mt/view?usp=drive_link

> To navigate to Fleet Control, go to **Capabilities → New Relic Control → Fleets**. Below is an example of a single fleet:

Reviewer 3: ```suggestion
To navigate to Fleet Control, go to **Capabilities → New Relic Control → Fleets**. You can use the pin to add New Relic Control to the left navigation panel for easy access.
```
Insert following image for illustration purposes: https://drive.google.com/file/d/1mq8oEi9fyljMZQsUrnADF8nDchO3I9mt/view?usp=drive_link

> 

Reviewer 3: Let's insert a screenshot here for illustrating the location of NR Control. https://drive.google.com/file/d/1mq8oEi9fyljMZQsUrnADF8nDchO3I9mt/view?usp=drive_link

## What is a fleet?
In Fleet Control, a fleet is a homogeneous collection of managed entities, such as Kubernetes clusters, that share identical instrumentation configurations. Fleets serve as logical groupings of managed entities, allowing for streamlined management and consistent observability configuration at scale.

A fleet is made up of three main components:
- **Active deployment:** The most recent deployment summary 
- **Configurations:** The configurations that are being applied to your Kubernetes clusters
- **Entities:** All the Kubernetes clusters within the fleet

> 

Reviewer 2: In this space please add this image:
https://drive.google.com/file/d/1q8ZiJblJ_dehSvOKdib453PPanwk-gY_/view?usp=sharing

### Understanding managed entities

> A managed entity in the context of Fleet Control refers to a Kubernetes cluster running New Relic infrastructure and logging instrumentation managed by Agent Control. Agent configuration is specified at the fleet level and every Kubernetes cluster in the fleet shares the same configuration as other clusters in that fleet. Note that a Kubernetes cluster cannot be added to a fleet unless Agent Control is installed on that cluster.

Reviewer 3: ```suggestion
A managed entity, in the context of Fleet Control, refers to a Kubernetes cluster running New Relic infrastructure and logging instrumentation managed by Agent Control. Agent configuration is specified at the fleet level and every Kubernetes cluster in the fleet shares the same configuration as other clusters in that fleet. 

[!IMPORTANT]
A Kubernetes cluster cannot be added to a fleet unless Agent Control is installed on that cluster.
```

- **Remote configuration:** Supports remote configuration of various agents. For details, see [scope](/docs/new-relic-control/getting-started/#scope). 

These agents are responsible for monitoring various aspects of your environment and reporting key data points, including metrics, events, logs, and traces (MELT).

## Prerequisites
Before setting up Fleet Control, ensure that your system meets the following prerequisites:

- **Opt in:** As Agent Control and Fleet Control are in a public preview, you have to opt in to to use the product. 
- **Agent Control:** Ensure you have installed [Agent Control](/docs/new-relic-control/agent-control/setup) on the Kubernetes clusters you plan to add to your fleets.

## Required permissions
- **Organization-scoped permissions:** The [Org Product Admin]() role (referred to as a Administrative setting in the product) is required to create and edit fleets within Fleet Control. Alternatively, you can [create a custom role](/docs/apis/nerdgraph/examples/nerdgraph-manage-groups/#create-role) to create/edit fleets.

> - **Fine-grained authorization:** To allow a group of users to deploy only a specific fleet:

Reviewer 2: ```suggestion
```

Reviewer 2: We decided not to include FGA for the Public Preview. Please remove this section.

>   - Ensure the users are not in a group with the [Org Product Admin]() role.

Reviewer 2: ```suggestion
```

>   - Select a group they belong to and assign the Fleet Manager role when creating or editing the fleet. This role is fleet-scoped. 

Reviewer 2: ```suggestion
```

>   - Or, create custom fleet-scoped roles using [the `customRoleCreate` API](/docs/apis/nerdgraph/examples/nerdgraph-manage-groups/#create-role).

Reviewer 2: ```suggestion
```

> <Callout variant="important">

Reviewer 2: ```suggestion
```

>     You must have the [Authentication Domain Manager](/docs/accounts/accounts-billing/new-relic-one-user-management/user-management-concepts/#admin-settings) role (referred to as Administrative settings in the product) to assign a group to a fleet.

Reviewer 2: ```suggestion
```

> </Callout>

Reviewer 2: ```suggestion
```

</Callout>

## Key features
- **Centralized management:** Offers a unified platform for administering agents running on fleets of Kubernetes clusters, allowing you to manage instrumentation across your infrastructure at scale.
- **Grouping and ownership:** Functions as both a grouping mechanism and an ownership boundary, enabling different teams to manage their respective fleets independently while maintaining consistent observability practices.
- **Remote configuration:** Supports remote configuration of various agents. For details, see [scope](/docs/new-relic-control/getting-started/#scope). 

These agents are responsible for monitoring various aspects of your environment and reporting key data points, including metrics, events, logs, and traces (MELT).

## Prerequisites
Before setting up Fleet Control, ensure that your system meets the following prerequisites:

- **Opt in:** As Agent Control and Fleet Control are in a public preview, you have to opt in to use the product. 
- **Agent Control:** Ensure you have installed [Agent Control](/docs/new-relic-control/agent-control/setup) on the Kubernetes clusters you plan to add to your fleets.

## Required permissions

> **Organization-scoped permissions:** The [Org Product Admin]() role (referred to as a Administrative setting in the product) is required to create and edit fleets within Fleet Control. Alternatively, you can [create a custom role](/docs/apis/nerdgraph/examples/nerdgraph-manage-groups/#create-role) with fleet management permissions.

Reviewer 2: ```suggestion
**Organization-scoped permissions:** The [Organization Manager]() role (referred to as a Administrative setting in the product) is required to create and edit fleets within Fleet Control.
```
